File: leetcode_python/lib/problems/p0026_remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        pointer = 0
        for i in range(0, len(nums)):
            if i > 0 and nums[pointer] != nums[i]:
                pointer += 1
                nums[pointer] = nums[i]

        return pointer + 1

File: leetcode_js/src/removeDuplicatesFromSortedArray.js
const removeDuplicates = function (nums) {
  let pointer = 0;
  for (let i = 0; i < nums.length; i++) {
    nums[pointer] = nums[i];
    pointer++;

    while (nums[i] === nums[i + 1]) {
      i++;
    }
  }

  return pointer;
};

module.exports = { removeDuplicates };
